client/src/client/client.go

```diff
@@ -53,8 +53,42 @@ func GetUserInput(text *string, reader io.Reader) error {
 // if any conn write call fails, return how many bytes you wrote and an error
 // if everything works out fine, return how many bytes you wrote and a nil error
 func SendToProducer(buf []byte, addr string) (int, error) {
-	// TODO
-	return 0, errors.New("Incomplete function")
+	// Opens a connection, if connection fails, return 0 and error
+	conn, err:= net.Dial("tcp", addr)
+	if err != nil {
+		conn.Close()
+		return 0, err
+	}
+	counter := 0
+	// While size of buf is > 0, send 1024 byte chunks to conn
```

Comment:
After we know `conn` is established successfully, I'd ask we use a `defer` statement since we're using Go and that's what it uses for resource cleanups/destructors. So basically you would only need to write `defer conn.Close()` once and it would run upon exiting the function, no matter where it returns.